internal/services/devcenter/dev_center_resource.go

```diff
@@ -234,5 +247,9 @@ func (r DevCenterResource) mapDevCenterToDevCenterResourceSchema(input devcenter
 
 	output.DevCenterUri = pointer.From(input.Properties.DevCenterUri)
 
+	if v := input.Properties.ProjectCatalogSettings; v != nil && v.CatalogItemSyncEnableStatus != nil {
+		output.ProjectCatalogItemSyncEnabled = pointer.From(input.Properties.ProjectCatalogSettings.CatalogItemSyncEnableStatus) == devcenters.CatalogItemSyncEnableStatusEnabled
+	}
```

Comment:
- Since we're assigning `input.Properties.ProjectCatalogSettings` to `v` we can just `v.CatalogItemSyncEnableStatus` instead, minor but just a bit shorter
- We can remove the nil check on `v.CatalogItemSyncEnableStatus` since we're using `pointer.From` which will handle that for us

```suggestion
	if v := input.Properties.ProjectCatalogSettings; v != nil {
		output.ProjectCatalogItemSyncEnabled = pointer.From(v.CatalogItemSyncEnableStatus) == devcenters.CatalogItemSyncEnableStatusEnabled
	}
```